src/components/ProgressIndicator/progressIndicator.spec.js

```diff
@@ -73,15 +75,37 @@ describe('The progress indicator component', () => {
     expect(stepPageItem).toBeVisible();
     const summaryPageItem = await screen.findByText('Summary');
     expect(summaryPageItem).toBeVisible();
-    const confirmationPageItem = await screen.findByText('Confirmation');
-    expect(confirmationPageItem).toBeVisible();
+  });
+
+  it('displays the available submission/form steps and hardcoded steps (with payment)', async () => {
+    mswServer.use(mockSubmissionPost(buildSubmission()));
+    const user = userEvent.setup({delay: null});
+    const form = getDefaultFactory({...FORM_DEFAULTS, paymentRequired: true})();
+
+    renderApp(form);
+
+    const startFormLink = await screen.findByRole('link', {name: 'Start page'});
+    user.click(startFormLink);
```

Comment:
`userEvent` is async, not applying this may lead to weird race conditions

```suggestion
    await user.click(startFormLink);
```